### PR TITLE
ttfb: init at 1.6.0

### PR DESCRIPTION
**Pull request by phip1611**
This adds ttfb to nixpkgs.

This is my first-ever upstreamed package.
I don't know what to set in the maintainers attribute.

###### Description of changes

Adds the CLI utility <https://github.com/phip1611/ttfb> to to nixpkgs.
It works on Linux and MacOS.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by JulienMalka — CHANGES_REQUESTED** [review]
Hi, thanks for your contribution ! You can register yourself as a maintainer by adding yourself to the ``maintainers/maintainer-list.nix`` file and then add yourself as a maintainer for this packet !

**Comment by phip1611** [comment]
> Hi, thanks for your contribution ! You can register yourself as a maintainer by adding yourself to the `maintainers/maintainer-list.nix` file and then add yourself as a maintainer for this packet !

Done. Once this is merged, I rebase https://github.com/NixOS/nixpkgs/pull/212639 for the new maintainer attribute.

**Review by JulienMalka — CHANGES_REQUESTED** [review]
Could you separate the 2 changes into 2 commits ?
``maintainers: add phip1611``
``ttfb: init at 1.6.0``
Thanks :)

**Comment by JulienMalka** [comment]
Otherwise this seems to be working well on ``x86_64-linux``.

**Comment by phip1611** [comment]
> Could you separate the 2 changes into 2 commits ? `maintainers: add phip1611` `ttfb: init at 1.6.0` Thanks :)

done @JulienMalka

**Review by JulienMalka — APPROVED** [review]
lgtm :)

File: pkgs/development/tools/ttfb/default.nix
Comment (SuperSandro2000) [inline pkgs/development/tools/ttfb/default.nix:23]: meta.description should be a string, not a mutliline string because search.nixos.org only handles one pine properly in the layout.

File: pkgs/development/tools/ttfb/default.nix
Comment (SuperSandro2000) [inline pkgs/development/tools/ttfb/default.nix:27]: longDescription should ideally not repeat description. Also it is optionally so you don't need to set it.

File: pkgs/development/tools/ttfb/default.nix
Comment (SuperSandro2000) [inline pkgs/development/tools/ttfb/default.nix:31]: It would be nice to set meta.changelog.

File: pkgs/development/tools/ttfb/default.nix
Comment (SuperSandro2000) [inline pkgs/development/tools/ttfb/default.nix:7]: Nit: normally we would move cargoSha256 after src but that's only stylistic.

File: pkgs/development/tools/ttfb/default.nix
Comment (SuperSandro2000) [inline pkgs/development/tools/ttfb/default.nix:20]:
```suggestion
  buildInputs = [ openssl ]
    ++ lib.optional student.isDarwin darwin.apple_sdk.frameworks.Security
```

**Comment by phip1611** [comment]
@SuperSandro2000 this is ready for another review round. Let's hope the darwin build works now.

**Comment by phip1611** [comment]
Oh no, stupid error. Gonna fix it on Monday.

File: pkgs/development/tools/ttfb/default.nix
Comment (NickCao) [inline pkgs/development/tools/ttfb/default.nix:34]:
```suggestion
    description = "CLI-Tool to measure the TTFB (time to first byte) of HTTP(S) requests";
```

File: pkgs/development/tools/ttfb/default.nix
Comment (NickCao) [inline pkgs/development/tools/ttfb/default.nix:40]:
```suggestion
    changelog = "https://github.com/phip1611/ttfb/blob/v${version}/CHANGELOG.md";
```

**Review by NickCao — APPROVED** [review]

**Pull request merged**
